core/src/identity.rs

```diff
@@ -0,0 +1,212 @@
+use serde::{Deserialize, Serialize};
+use signature::keypair::SSIKeyMaterial;
+
+use signature::keypair::Ed25519SSIKeyPair;
```

Comment:
We should not be using the specific implementation but instead the trait

Reply:
It's trait, but Since the KeyPair trait require public and private traits, we still have to maintain Ed25519SSIKeyPair or any other *KeyPair trait. There is still separation from Ed25519Signer traits.